owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/files/RemoteFile.kt

```diff
@@ -0,0 +1,153 @@
+    }
+
+    /**
+     * Use this to find out if this file is a folder.
+     *
+     * @return true if it is a folder
+     */
+    val isFolder
+        get() = mimeType == MIME_DIR || mimeType == MIME_DIR_UNIX
+
+    companion object {
+
+        const val MIME_DIR = "DIR"
+        const val MIME_DIR_UNIX = "httpd/unix-directory"
+
+        fun getRemoteFileFromDav(davResource: Response, userId: String, userName: String): RemoteFile {
+            val remotePath = getRemotePathFromUrl(davResource.href, userId)
+            val remoteFile = RemoteFile(remotePath = remotePath, owner = userName)
+            val properties = davResource.properties
+
+            for (property in properties) {
+                when (property) {
+                    is CreationDate -> {
+                        remoteFile.creationTimestamp = property.creationDate.toLong()
+                    }
+                    is GetContentLength -> {
+                        remoteFile.length = property.contentLength
+                    }
+                    is GetContentType -> {
+                        property.type?.let { remoteFile.mimeType = it }
+                    }
+                    is GetLastModified -> {
+                        remoteFile.modifiedTimestamp = property.lastModified
+                    }
+                    is GetETag -> {
+                        remoteFile.etag = property.eTag
+                    }
+                    is OCPermissions -> {
+                        remoteFile.permissions = property.permission
+                    }
+                    is OCId -> {
+                        remoteFile.remoteId = property.id
+                    }
+                    is OCSize -> {
+                        remoteFile.size = property.size
+                    }
+                    is QuotaUsedBytes -> {
+                        remoteFile.quotaUsedBytes = BigDecimal.valueOf(property.quotaUsedBytes)
+                    }
+                    is QuotaAvailableBytes -> {
+                        remoteFile.quotaAvailableBytes = BigDecimal.valueOf(property.quotaAvailableBytes)
+                    }
+                    is OCPrivatelink -> {
+                        remoteFile.privateLink = property.link
+                    }
+                }
+            }
+            return remoteFile
+        }
+
+        /**
+         * Retrieves a relative path from a remote file url
+         *
+         *
+         * Example: url:port/remote.php/dav/files/username/Documents/text.txt => /Documents/text.txt
+         *
+         * @param url    remote file url
+         * @param userId file owner
+         * @return remote relative path of the file
+         */
+        private fun getRemotePathFromUrl(url: HttpUrl, userId: String): String {
+            val davFilesPath = OwnCloudClient.WEBDAV_FILES_PATH_4_0 + userId
+            val absoluteDavPath = Uri.decode(url.encodedPath)
+            val pathToOc = absoluteDavPath.split(davFilesPath)[0]
```

Comment:
`absoluteDavPath.split(davFilesPath).first()?`

Reply:
Basically the same, no problem with it.